core/exception/backtrace_locations_spec.rb

```diff
@@ -36,4 +36,21 @@
       e.backtrace_locations[0].should == "backtrace first"
     end
   end
+
+  it "includes the nesting level of a block as part of the backtrace label" do
+    first_level_location, second_level_location, third_level_location =
+      ExceptionSpecs::Backtrace.backtrace_locations_inside_nested_blocks
+
+    first_level_location.label.should == 'block in backtrace_locations_inside_nested_blocks'
+    second_level_location.label.should == 'block (2 levels) in backtrace_locations_inside_nested_blocks'
+    third_level_location.label.should == 'block (3 levels) in backtrace_locations_inside_nested_blocks'
+  end
+
+  it "sets the backtrace label for a top-level block differently depending on it being in the main file or a required file" do
+    path = fixture(__FILE__, "backtrace_locations_in_main.rb")
+    main_label, required_label = ruby_exe(path).lines
+
+    main_label.strip.should == 'block in <main>'
+    required_label.strip.should == 'block in <top (required)>'
+  end
```

Comment:
Could you move those specs to spec/ruby/core/thread/backtrace/location/label_spec.rb ?
That's where other specs for `Thread::Backtrace::Location#label` are.

Reply:
It's OK to keep using `Thread.current.backtrace_locations` instead of `caller_locations`, they should be identical.

Reply:
👍